src/appHost.ts

```diff
@@ -486,7 +486,11 @@ miss: ${memoizedWithMissHit.miss}
 
     function getAPI<TAPI>(key: SlotKey<TAPI>): TAPI {
         const APISlot = getSlot<TAPI>(key)
-        return APISlot.getSingleItem().contribution
+        const item = APISlot.getSingleItem()
+        if (item) {
+            return item.contribution
+        }
+        throw new Error(`Extension item for extension slot with key '${key}' doesn't exist.`)
     }
```

Comment:
@itsh01 also missed that one. let me know what do you think of this change, i'm not sure if exception is the best way to solve this. Should this even happen?

Reply:
In practice, that will never happen for calls from shell, since it's verified that they have the API in the dependency APIs, and they are not loaded until all dependency APIs are ready.
However, if `getAPI` is called directly on the host and the API doesn't exist it's probably make sense to throw.
Let's just change the error to be more explicit with the error text (`API '${slotKeyToName(key)}' doesn't exist.`)

Reply:
got it. updated.